webrob/pages/neems.py

```diff
@@ -1,16 +1,30 @@
-from flask import session, send_from_directory, jsonify, request, render_template
+from flask import session, send_from_directory, jsonify, request, render_template, redirect, url_for, flash
 from flask_paginate import Pagination, get_page_args
 
-from webrob.app_and_db import app
+from webrob.app_and_db import app, get_mongo_db_meta_collection
+from pymongo.errors import ConnectionFailure, PyMongoError
+
 
 from webrob.neems.manager import NEEM_Manager
+from webrob.models.NEEMMetaException import NEEMMetaException
+
 
 __author__ = 'danielb@cs.uni-bremen.de'
 
 neem_manager = NEEM_Manager()
 
 @app.route('/neems')
 def render_neems():
+
+    # at first check if there is settings stored in db
+    try:
+        mongoDBMetaCollection = get_mongo_db_meta_collection()
+    except ConnectionFailure as e:
+        raise NEEMMetaException('An exception has occurred during connection with mongodb collection, please check!', exc=e)
+
+    except PyMongoError as e:
```

Comment:
why not adding a flask handler for this error?

Reply:
done. Now PyMongoError has its own flask handler.